src/query/storages/system/src/tables_table.rs

```diff
@@ -113,11 +113,20 @@ where TablesTable<T>: HistoryAware
                 let name: &str = Box::leak(name);
                 let tables = match Self::list_tables(&ctl, tenant.as_str(), name).await {
                     Ok(tables) => tables,
-                    Err(err) if err.code() == ErrorCode::EMPTY_SHARE_ENDPOINT_CONFIG => {
-                        tracing::warn!("list tables failed on db {}: {}", db.name(), err);
-                        continue;
+                    Err(err) => {
+                        // Swallow the errors related with sharing. Listing tables in a shared database
+                        // is easy to get errors with invalid configs, but system.tables should not be
+                        // affected by it.
+                        if db.get_db_info().meta.from_share.is_some() {
```

Comment:
I dunno is this a feasible approach to check the database is created from sharing or not, @lichuang any suggestion?

Reply:
yes, when `db_info.from_share` is_some, it means that this db is created from a share